src/compiler/parser.ts

```diff
@@ -5883,13 +5884,24 @@ namespace ts {
                             amdDependencies.push(amdDependency);
                         }
                     }
+
+                    const checkJsDirectiveRegEx = /^\/\/\/?\s*(@ts-check|@ts-nocheck)\s*$/gim;

```

Comment:
I would recommend you pull this out of the function and remove the `g` option. `g` is only really useful if you want to preserve `lastIndex` when using a regular expression in a loop. Also I would remove the `m` option as it should be unnecessary. 

Consider this instead:
```ts
const checkJsDirectiveRegEx = /^\/\/\/?\s*@ts-(no)?check\s*$/i;
...
const checkJsDirectiveMatchResult = checkJsDirectiveRegEx.exec(comment);
if (checkJsDirectiveMatchResult) {
  checkJsDirective = {
    enabled: !!checkJsDirectiveMatchResult[1],
    end: range.end,
    pos: range.pos
  };
}
```

Capturing only the `no` allows you to avoid the additional case-insensitive string comparison.